js/modules/k6/experimental/fs/module_test.go

```diff
@@ -212,6 +212,196 @@ func TestFile(t *testing.T) {
 
 		assert.NoError(t, err)
 	})
+
+	t.Run("read in multiple iterations", func(t *testing.T) {
+		t.Parallel()
+
+		runtime, err := newConfiguredRuntime(t)
+		require.NoError(t, err)
+
+		testFilePath := fsext.FilePathSeparator + "bonjour.txt"
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, testFilePath, []byte("hello"), 0o644)
+		})
+		runtime.VU.InitEnvField.FileSystems["file"] = fs
+
+		_, err = runtime.RunOnEventLoop(wrapInAsyncLambda(fmt.Sprintf(`
+			const file = await fs.open(%q);
+
+			let fileContent = new Uint8Array(5);
+
+			let bytesRead;
+			let buffer = new Uint8Array(3);
+
+			bytesRead = await file.read(buffer)
+			if (bytesRead !== 3) {
+				throw 'expected read to return 3, got ' + bytesRead + ' instead';
+			}
+
+			fileContent.set(buffer, 0);
+
+			bytesRead = await file.read(buffer)
+			if (bytesRead !== 2) {
+				throw 'expected read to return 2, got ' + bytesRead + ' instead';
+			}
+
+			fileContent.set(buffer.subarray(0, bytesRead), 3);
+
+			bytesRead = await file.read(buffer)
+			if (bytesRead !== null) {
+				throw 'expected read to return null, got ' + bytesRead + ' instead';
+			}
+		`, testFilePath)))
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("read called with invalid argument should fail", func(t *testing.T) {
+		t.Parallel()
+
+		runtime, err := newConfiguredRuntime(t)
+		require.NoError(t, err)
+
+		testFilePath := fsext.FilePathSeparator + "bonjour.txt"
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, testFilePath, []byte("Bonjour, le monde"), 0o644)
+		})
+		runtime.VU.InitEnvField.FileSystems["file"] = fs
+
+		_, err = runtime.RunOnEventLoop(wrapInAsyncLambda(fmt.Sprintf(`
+			const file = await fs.open(%q);	
+			let bytesRead;
+
+			// No argument should fail with TypeError.
+			bytesRead = await file
+				.read()
+				.catch(err => {
+					if (err.name !== 'TypeError') {
+						throw 'unexpected error: ' + err;
+					}
+				})
+
+			// null buffer argument should fail with TypeError.
+			bytesRead = await file
+				.read(null)
+				.catch(err => {
+					if (err.name !== 'TypeError') {
+						throw 'unexpected error: ' + err;
+					}
+				})
+
+			// undefined buffer argument should fail with TypeError.
+			bytesRead = await file
+				.read(undefined)
+				.catch(err => {
+					if (err.name !== 'TypeError') {
+						throw 'unexpected error: ' + err;
+					}
+				})
```

Comment:
nit: just for a comparision 

```suggestion
			try { 
				bytesRead = await file.read(undefined)
			} catch (err)  {
				if (err.name !== 'TypeError') {
					throw 'unexpected error: ' + err;
				}
			}
```

For all my dislike of exceptions, this IMO looks just as good if not better and IMO a lot more what most people will write. I actually am interested why you didn't write it like that? You don't like how it works? You thought it won't work?

Reply:
I see what you mean. I wrote it this way out of personal preference and curiosity for this notation way I didn't know yet.

Seeing it compared like this in hindsight, I think I agree that the `try/catch` block might indeed be easier to read and decipher 👍🏻

Reply:
I've replaced the relevant test code with `try/catch` blocks indeed 👍🏻